# Structural Analysis of APC (Wild-Type vs R876* Mutant)
 - This document describes the structural analysis of the APC protein, comparing the wild-type and R876* nonsense mutant.
 - The analysis focuses on:
   - Homology modeling of APC
   - Visualization of protein truncation caused by R876*
   - Structural comparison between wild-type and mutant
   - Identification of a predicted drug-binding pocket using surface analysis

 ## Objective
 - Model APC wild-type and R876* mutant structures
 - Analyze the structural consequence of a truncating mutation
 - Visualize loss of downstream domains
 - Identify and highlight a potential drug-binding pocket

## Homology Modeling
 - Homology models for APC wild-type and R876* mutant were generated using **SWISS-MODEL**.
   - Template used: PDB ID 4YK6.1
   - Method: X-ray crystallography
   - Resolution: 1.7 Å
   - Coverage: Partial APC domain (not full-length protein)

 ### Important Note
 - The R876* mutation introduces a premature stop codon.
 - Since the modeled structure represents only a partial domain (~289 residues) the mutation site itself is not present in the structure.
 - Therefore structural interpretation focuses on truncation effects and domain loss, not local residue substitution.
---

## Software and Input Files

 **Software Used**: PyMOL Molecular Graphics System
   
 **Input PDB Files**: normal.pdb → APC Wild-Type, Mutated.pdb → APC R876* Mutant

### I.Loading Structures into PyMOL
 **1. Load PDB files**
 ```
 load normal.pdb
 load Mutated.pdb
```
**2. Confirm loaded objects**
```
ls
```
**3. Expected output**
```
normal
Mutated
```
### II.Initial Visualization
 **1. Reset and prepare visualization**
 ```
hide everything
show cartoon, normal
show cartoon, Mutated
```
 **2. Color structures**
 ```
color cyan, normal
color grey, Mutated
```
**3. Viewing both structure in same display**
```
translate [50, 0, 0], mutated
```
**Purpose**:To confirm that both wild-type and mutant structures are correctly loaded and visually comparable.

![Structural Analysis of APC](./images/APC%20normal%20and%20mutated.png)

### III.Structural Alignment (Wild-Type vs Mutant)
 **1. Align mutant to wild-type**
 ```
align Mutated, normal
```
**2. Expected output:**
```
Executive: RMSD = 0.000
```
**Interpretation**:
 - Although the mutant APC structure contains a premature truncation, alignment was performed to compare the shared modeled region (residues 1–289) between wild-type and mutant structures.
 - Wild-type and mutant structures are **structurally identical within the modeled region**
 - Expected outcome since both models **use the same template**
 - Confirms absence of global backbone deviation

 ![Structural Analysis of APC](./images/APC%20aligning.png)

### Understanding the R876* Mutation
 - R876* is a **nonsense mutation**
 - Causes **premature termination** of APC translation
 - Results in **loss of downstream functional domains**
 - Modeled structure contains only **residues 1–289**
 - Correct strategy: Visualize the C-terminal end of the modeled fragment to represent truncation.

### IV.Identifying the Residue Range
```
iterate all and name CA, print(resi)
```
**Observed range:**
```
Residues 1 to 289
```
### V.Truncation Visuvalization in Wild-Type vs mutant APC
```
select trunc_end, mutated and resi 270-289
show sticks, trunc_end
color red, trunc_end
```
![Structural Analysis of APC](./images/APC%20truncation%201.png)
- The wild-type structure is shown in **cyan**, the mutant structure in **grey**, and the C-terminal truncation region (residues 270–289) in **red sticks**, highlighting the premature termination caused by the R876* mutation.

### VI.Visualizing Protein Truncation
Following structural alignment, the mutant APC structure was analyzed independently to visualize the structural consequence of the R876* truncation.
**1.Select C-terminal region**
```
select trunc_end, Mutated and resi 270-289
```
**2.Highlight truncation region**
```
show sticks, trunc_end
color red, trunc_end
```
**3.Clean visualization**
```
hide everything
show cartoon, Mutated
color grey, Mutated
show sticks, trunc_end
color red, trunc_end
zoom Mutated, 15
```
**Interpretation**
 - Red sticks represent the **last structured residues before truncation**
 - All downstream regions are **absent due to R876**

 ![Structural Analysis of APC](./images/APC%20aligning%20zoom.png)
   

**Final Interpretation**
 - The R876* mutation causes **truncation of APC**
 - Downstream functional **domains are lost**
 - A structured domain containing a **predicted surface pocket remains**
 - The pocket is spatially distinct from the truncation boundary
 - This suggests that **residual APC domains may remain structurally targetable**, despite loss of tumor suppressor function.

### Summary
 - APC truncation was analyzed using **domain-level structural reasoning**
 - PyMOL commands enabled visualization of truncation and surface features
 - Structural findings align with known biological consequences of APC loss
 - This analysis demonstrates how truncating mutations can be interpreted structurally even without full-length protein structures
